File: examples/swarm-research-task.js
import { DashClaw } from '../sdk/dashclaw.js';
import dotenv from 'dotenv';
dotenv.config();

const claw = new DashClaw({
  baseUrl: process.env.DASHCLAW_BASE_URL || 'http://localhost:3000',
  apiKey: process.env.DASHCLAW_API_KEY,
  agentId: 'swarm-orchestrator',
});

async function main() {
  const swarmId = 'swarm-node-test-1';
  console.log(`Starting swarm orchestration: ${swarmId}`);

  // 1. Parent Task
  const { action: parent } = await claw.createAction({
    action_type: 'orchestration',
    declared_goal: 'Fulfill customer support request #9921',
    swarm_id: swarmId,
    risk_score: 45
  });

  const parentId = parent.action_id;
  console.log(`Parent Action Created: ${parentId}`);

  // 2. Sub-task: Knowledge lookup
  const { action: sub1 } = await claw.createAction({
    action_type: 'lookup',
    declared_goal: 'Find billing info for User #122',
    parent_action_id: parentId,
    swarm_id: swarmId,
    systems_touched: ['stripe', 'internal_db']
  });

  console.log(`Sub-action 1 (lookup): ${sub1.action_id}`);

  // 3. Sub-task: Response drafting
  const { action: sub2 } = await claw.createAction({
    action_type: 'drafting',
    declared_goal: 'Compose apology and billing update',
    parent_action_id: parentId,
    swarm_id: swarmId
  });

  console.log(`Sub-action 2 (draft): ${sub2.action_id}`);

  // 4. Resolve Parent
  await claw.updateOutcome(parentId, {
    status: 'completed',
    output_summary: 'Knowledge retrieved and draft composed. Task complete.'
  });

  console.log('Orchestration finished.');
}

main();

File: examples/swarm-research-task.py
import os
import asyncio
from dashclaw import DashClaw
from dotenv import load_dotenv

load_dotenv()

claw = DashClaw(
    base_url=os.environ.get("DASHCLAW_BASE_URL", "http://localhost:3000"),
    api_key=os.environ["DASHCLAW_API_KEY"],
    agent_id="research-coordinator",
    agent_name="Research Coordinator",
)

def main():
    # 1. High-level COORDINATOR action
    parent = claw.create_action(
        action_type="workflow",
        declared_goal="Analyze market trends for AI safety",
        swarm_id="swarm-alpha-9",
        risk_score=40
    )
    parent_id = parent.get("action_id")
    print(f"Parent Action Created: {parent_id}")

    # 2. Sub-action: Data Collection
    child_1 = claw.create_action(
        action_type="research",
        declared_goal="Scrape recent whitepapers",
        parent_action_id=parent_id,
        swarm_id="swarm-alpha-9",
        systems_touched=["arxiv", "google_scholar"]
    )
    print(f"Child Action 1 Created: {child_1.get('action_id')}")

    # 3. Sub-action: Analysis
    child_2 = claw.create_action(
        action_type="analysis",
        declared_goal="Summarize findings",
        parent_action_id=parent_id,
        swarm_id="swarm-alpha-9",
        confidence=92
    )
    print(f"Child Action 2 Created: {child_2.get('action_id')}")

    # 4. Finalize Parent
    claw.update_outcome(parent_id, {
        "status": "completed",
        "output_summary": "Research workflow finished with 2 sub-tasks completed."
    })
    print("Workflow complete.")

if __name__ == "__main__":
    main()
